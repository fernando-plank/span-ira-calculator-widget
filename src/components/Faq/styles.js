import styled from 'styled-components';
export const Wrapper = styled.div `
  padding: 30px 19px;
  width: 100%;
  color: ${({ theme }) => theme.colors.black};
  text-align: left;
`;
export const Title = styled.h1 `
  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    font-weight: 700;
    font-size: 48px;
    line-height: 56px;
    margin-bottom: 46px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    font-weight: 700;
    font-size: 48px;
    line-height: 56px;
    margin-bottom: 88px;
  }
`;
export const FaqBody = styled.div `
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
`;
