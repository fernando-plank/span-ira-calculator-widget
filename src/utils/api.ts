import { IncentivesResponse, TaxCreditsIncentives } from '@services/types'
import { makeTaxCreditsTooltip } from './tooltips'

export const formatTaxCreditsIncentives = (
  tax_credit_incentives: TaxCreditsIncentives[]
): TaxCreditsIncentives[] => {
  let eletricPanel: any = null
  let items: TaxCreditsIncentives[] = tax_credit_incentives.filter((item) => {
    if (item.item === 'Electric Panel') {
      eletricPanel = item
    }
    return item.type === 'tax_credit' && item.item !== 'Electric Panel'
  })

  items = items?.filter(
    (item, index, self) =>
      index ===
      self.findIndex((t) => {
        return t.item === item.item
      })
  )

  if (items && eletricPanel) {
    eletricPanel.amount = 600
    eletricPanel['item'] = 'SPAN Smart Electrical Panel'
    items.unshift(eletricPanel)
    items = makeTaxCreditsTooltip(items)
  }

  return items
}
