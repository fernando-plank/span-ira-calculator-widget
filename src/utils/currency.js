export const formatCurrency = (number) => {
    if (!number)
        return 'N/A';
    return Intl.NumberFormat('EN-US', {
        style: 'currency',
        maximumSignificantDigits: 21,
        currency: 'USD'
    }).format(number);
};
