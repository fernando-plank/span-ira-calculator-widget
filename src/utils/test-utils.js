var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
/* eslint-disable import/export */
import React from 'react';
import theme from '@styles/theme';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
jest.mock('../components/Icon', () => {
    return jest.fn(() => React.createElement("div", null, "mocked icon"));
});
const customRender = (ui, _a = {}) => {
    var renderOptions = __rest(_a, []);
    return render(React.createElement(ThemeProvider, { theme: theme }, ui), renderOptions);
};
export * from '@testing-library/react';
export { customRender as render };
