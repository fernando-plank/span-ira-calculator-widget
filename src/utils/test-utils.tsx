/* eslint-disable import/export */
import { ReactElement } from 'react'
import { RenderOptions, render } from '@testing-library/react'

import theme from '@styles/theme'
import { ThemeProvider } from 'styled-components'

jest.mock('../components/Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

const customRender = (ui: ReactElement, { ...options }: RenderOptions = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options)

export * from '@testing-library/react'
export { customRender as render }
