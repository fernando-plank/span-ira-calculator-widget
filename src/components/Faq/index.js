import React from 'react';
import Accordion from '@components/Accordion';
import * as S from './styles';
const Faq = () => {
    const accordionStyle = { fontSize: '16px', lineHeight: 'normal' };
    return (React.createElement(S.Wrapper, null,
        React.createElement(S.Title, { "test-dataid": "faq-component-title" }, "Rebate FAQs"),
        React.createElement(Accordion, { style: accordionStyle, icon: 'Plus', title: "Will Enphase IQ  integrate with SPAN without SPAN Remote Meter?" },
            React.createElement(S.FaqBody, null, "All tax credits are available now. The rebates should start to become available by the end of 2023. We don`\u2018`t know exactly when the rebates marked \u201Clate 2023\u201D will be available, because it will depend on how each state rolls out its incentive program.")),
        React.createElement(Accordion, { style: accordionStyle, icon: 'Plus', title: "What is SPAN Remote Meter?", "data-testid": "accordion-span-remote-meter" },
            React.createElement(S.FaqBody, null, "In many cases, no! The IRA\u2019s incentives are designed to increase access to clean technology. For households with lower incomes, up to 100% of appliance and installation costs are discounted at purchase, meaning you could install efficient electric appliances at no cost, with no spending.Middle-income households do have to spend in order to access savings, but up to 50% of appliance and installation costs can be covered through upfront discounts, and you can use tax credits to cover some of the remaining gap. Highest-income households are not eligible for upfront discounts, so you will have to pay full price for appliances and installation \u2014 but tax credits on the back end could recoup up to 30% of your costs.")),
        React.createElement(Accordion, { style: accordionStyle, icon: 'Plus', title: "How does it work?" },
            React.createElement(S.FaqBody, null,
                React.createElement("p", null, "Yes! The IRA\u2019s up-front electrification rebates and electrification tax credits can all be used by renters. Many of these upgrades \u2014 including window-unit heat pumps (which should qualify by 2024/2025), electric stoves/cooktops, and heat pump clothes dryers \u2014 are portable, so renters can bring them to their next homes and won\u2019t have to leave any savings behind. Renters are also eligible for the used and new EV tax credits. And although they\u2019re not exactly consumer-facing, renters will benefit from the IRA\u2019s $1 billion investment in affordable housing energy upgrades and the new tax deduction for efficiency upgrades in commercial buildings (including apartment buildings). Renters can also subscribe to community solar \u2014 which will be cheaper because of the IRA\u2019s supply-side renewable energy incentives."))),
        React.createElement(Accordion, { style: accordionStyle, icon: 'Plus', title: "Where is the Remote Meter installed?" },
            React.createElement(S.FaqBody, null, "All tax credits were available starting January 1, 2023, so save your receipts for filing season! If you installed rooftop solar in 2022, it is eligible for the new, 30% 25D tax credit. If you installed a heat pump in 2022, it is eligible for the old 25C tax credit: 10% of costs up to $300. The Electrification Rebates and Efficiency Rebates will likely not be retroactive."))));
};
export default Faq;
