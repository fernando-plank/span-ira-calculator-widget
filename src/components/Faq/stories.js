import React from 'react';
import Faq from '.';
export default {
    title: 'Faq',
    component: Faq
};
export const Default = () => React.createElement(Faq, null);
