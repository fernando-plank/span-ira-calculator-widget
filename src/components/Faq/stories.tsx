import React from 'react'
import { Story, Meta } from '@storybook/react'

import Faq from '.'

export default {
  title: 'Faq',
  component: Faq
} as Meta

export const Default: Story = () => <Faq />
