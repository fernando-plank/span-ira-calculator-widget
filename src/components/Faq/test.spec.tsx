import { screen } from '@testing-library/react'
import React from 'react'
import { render } from '../../utils/test-utils'

import Faq from '.'

jest.mock('../Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

describe('<Faq />', () => {
  it('should render the title', () => {
    const { container } = render(<Faq />)

    expect(screen.getByText('Rebate FAQs')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
