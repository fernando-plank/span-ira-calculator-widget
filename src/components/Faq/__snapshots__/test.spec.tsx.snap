// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Faq /> should render the title 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  cursor: pointer;
}

align-items:baseline .c2 > svg {
  cursor: pointer;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  margin-bottom: 17px;
  z-index: 2;
  overflow: hidden;
  position: relative;
  -webkit-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;
  max-height: 0px;
}

.c9.active {
  max-height: 1000px;
}

.c5 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c7 {
  width: 2px;
  height: 24px;
  background-color: #1f1f1f;
  border-radius: 2px;
  position: absolute;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c7.active {
  -webkit-transform: translate3d(0px,0px,0px) scale3d(1,1,1) rotateX(0deg) rotateY(0deg) rotateZ(90deg) skew(0deg,0deg);
  -ms-transform: translate3d(0px,0px,0px) scale3d(1,1,1) rotateX(0deg) rotateY(0deg) rotateZ(90deg) skew(0deg,0deg);
  transform: translate3d(0px,0px,0px) scale3d(1,1,1) rotateX(0deg) rotateY(0deg) rotateZ(90deg) skew(0deg,0deg);
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
}

.c6 {
  width: 24px;
  height: 2px;
  background-color: #1f1f1f;
  border-radius: 2px;
}

.c4 {
  font-style: normal;
  font-weight: 700;
  font-size: 32px;
  line-height: 40px;
  width: 100%;
}

.c8 {
  height: 1px;
  background: #b0b0b1;
  border-radius: 2px;
  margin-bottom: 17px;
}

.c0 {
  padding: 30px 19px;
  width: 100%;
  color: #1F1F1F;
  text-align: left;
}

.c10 {
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
}

@media (min-width:320px) {
  .c9 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:1024px) {
  .c9 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:320px) {
  .c1 {
    font-weight: 700;
    font-size: 48px;
    line-height: 56px;
    margin-bottom: 46px;
  }
}

@media (min-width:1024px) {
  .c1 {
    font-weight: 700;
    font-size: 48px;
    line-height: 56px;
    margin-bottom: 88px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Rebate FAQs
  </h1>
  <div
    class="c2 c3"
  >
    <h2
      class="c4"
      data-testid="accordion-title"
      style="font-size: 16px; line-height: normal;"
    >
      Will Enphase IQ  integrate with SPAN without SPAN Remote Meter?
    </h2>
    <div
      class="c5"
    >
      <div
        class="c6"
      />
      <div
        class="c7 accordion-body"
      />
    </div>
  </div>
  <div
    class="c8"
  />
  <div
    class="c9 accordion-body"
    data-testid="accordion-body"
  >
    <div
      class="c10"
    >
      All tax credits are available now. The rebates should start to become available by the end of 2023. We don\`‘\`t know exactly when the rebates marked “late 2023” will be available, because it will depend on how each state rolls out its incentive program.
    </div>
  </div>
  <div
    class="c2 c3"
  >
    <h2
      class="c4"
      data-testid="accordion-title"
      style="font-size: 16px; line-height: normal;"
    >
      What is SPAN Remote Meter?
    </h2>
    <div
      class="c5"
    >
      <div
        class="c6"
      />
      <div
        class="c7 accordion-body"
      />
    </div>
  </div>
  <div
    class="c8"
  />
  <div
    class="c9 accordion-body"
    data-testid="accordion-body"
  >
    <div
      class="c10"
    >
      In many cases, no! The IRA’s incentives are designed to increase access to clean technology. For households with lower incomes, up to 100% of appliance and installation costs are discounted at purchase, meaning you could install efficient electric appliances at no cost, with no spending.Middle-income households do have to spend in order to access savings, but up to 50% of appliance and installation costs can be covered through upfront discounts, and you can use tax credits to cover some of the remaining gap. Highest-income households are not eligible for upfront discounts, so you will have to pay full price for appliances and installation — but tax credits on the back end could recoup up to 30% of your costs.
    </div>
  </div>
  <div
    class="c2 c3"
  >
    <h2
      class="c4"
      data-testid="accordion-title"
      style="font-size: 16px; line-height: normal;"
    >
      How does it work?
    </h2>
    <div
      class="c5"
    >
      <div
        class="c6"
      />
      <div
        class="c7 accordion-body"
      />
    </div>
  </div>
  <div
    class="c8"
  />
  <div
    class="c9 accordion-body"
    data-testid="accordion-body"
  >
    <div
      class="c10"
    >
      <p>
        Yes! The IRA’s up-front electrification rebates and electrification tax credits can all be used by renters. Many of these upgrades — including window-unit heat pumps (which should qualify by 2024/2025), electric stoves/cooktops, and heat pump clothes dryers — are portable, so renters can bring them to their next homes and won’t have to leave any savings behind. Renters are also eligible for the used and new EV tax credits. And although they’re not exactly consumer-facing, renters will benefit from the IRA’s $1 billion investment in affordable housing energy upgrades and the new tax deduction for efficiency upgrades in commercial buildings (including apartment buildings). Renters can also subscribe to community solar — which will be cheaper because of the IRA’s supply-side renewable energy incentives.
      </p>
    </div>
  </div>
  <div
    class="c2 c3"
  >
    <h2
      class="c4"
      data-testid="accordion-title"
      style="font-size: 16px; line-height: normal;"
    >
      Where is the Remote Meter installed?
    </h2>
    <div
      class="c5"
    >
      <div
        class="c6"
      />
      <div
        class="c7 accordion-body"
      />
    </div>
  </div>
  <div
    class="c8"
  />
  <div
    class="c9 accordion-body"
    data-testid="accordion-body"
  >
    <div
      class="c10"
    >
      All tax credits were available starting January 1, 2023, so save your receipts for filing season! If you installed rooftop solar in 2022, it is eligible for the new, 30% 25D tax credit. If you installed a heat pump in 2022, it is eligible for the old 25C tax credit: 10% of costs up to $300. The Electrification Rebates and Efficiency Rebates will likely not be retroactive.
    </div>
  </div>
</div>
`;
